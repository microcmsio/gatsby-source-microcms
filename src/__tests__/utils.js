const {
  makeTypeName,
  isListContent,
  isObjectContent,
  createContentNode,
} = require('../utils');

const createNode = jest.fn();
const createNodeId = jest.fn();

beforeEach(() => {
  createNode.mockClear();
  createNodeId.mockClear();
});

test('makeTypeName', () => {
  expect(makeTypeName('type')).toBe('MicrocmsType');
  expect(makeTypeName('Type')).toBe('MicrocmsType');
});

describe('content type', () => {
  test('isListContent', () => {
    expect(isListContent({ format: 'list', content: [] })).toBe(true);
    expect(isListContent({ format: 'list', content: {} })).toBe(false);
    expect(isListContent({ format: 'invalid', content: [] })).toBe(false);
  });

  test('isObjectContent', () => {
    expect(
      isObjectContent({
        format: 'object',
        content: {
          id: 'id',
        },
      })
    ).toBe(true);
    expect(isObjectContent({ format: 'object', content: [] })).toBe(false);
    expect(
      isObjectContent({
        format: 'invalid',
        content: {
          id: 'id',
        },
      })
    ).toBe(false);
  });
});

describe('createContentNode', () => {
  test('content id', () => {
    createNodeId.mockReturnValue('nodeId');
    createContentNode({
      createNode,
      createNodeId,
      sortIndex: 0,
      content: { id: 'abcde' },
      type: 'post',
    });
    expect(createNode.mock.calls).toMatchSnapshot();
    expect(createNode.mock.calls[0][0]).not.toHaveProperty('abcdeId');
  });

  test('content body', () => {
    createNodeId.mockReturnValue('nodeId');
    createContentNode({
      createNode,
      createNodeId,
      sortIndex: 0,
      content: { id: 'vwxyz', body: 'content' },
      type: 'news',
    });
    expect(createNode.mock.calls).toMatchSnapshot();
    expect(createNode.mock.calls[0][0]).not.toHaveProperty('vwxyzId');
  });

  test('content does not have id', () => {
    createNodeId.mockReturnValue('nodeId');
    createContentNode({
      createNode,
      createNodeId,
      sortIndex: 0,
      content: { body: 'content' },
      type: 'body',
    });
    expect(createNode.mock.calls).toMatchSnapshot();
    expect(createNode.mock.calls[0][0]).not.toHaveProperty('bodyId');
  });
});
