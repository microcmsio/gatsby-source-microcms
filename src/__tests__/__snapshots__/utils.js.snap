// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`createContentNode content body 1`] = `
Array [
  Array [
    Object {
      "body": "content",
      "children": Array [],
      "id": "nodeId",
      "internal": Object {
        "content": "{\\"id\\":\\"vwxyz\\",\\"body\\":\\"content\\"}",
        "contentDigest": "dc80de2bd37db677dea84c9626b2aef0",
        "type": "MicrocmsNews",
      },
      "newsId": "vwxyz",
      "parent": null,
      "sortIndex": 0,
    },
  ],
]
`;

exports[`createContentNode content does not have id 1`] = `
Array [
  Array [
    Object {
      "body": "content",
      "children": Array [],
      "id": "nodeId",
      "internal": Object {
        "content": "{\\"body\\":\\"content\\"}",
        "contentDigest": "067d01e67619fda073f1510f2d8811ce",
        "type": "MicrocmsBody",
      },
      "parent": null,
      "sortIndex": 0,
    },
  ],
]
`;

exports[`createContentNode content id 1`] = `
Array [
  Array [
    Object {
      "children": Array [],
      "id": "nodeId",
      "internal": Object {
        "content": "{\\"id\\":\\"abcde\\"}",
        "contentDigest": "d816d8eec404705cf80f72fe03d6834f",
        "type": "MicrocmsPost",
      },
      "parent": null,
      "postId": "abcde",
      "sortIndex": 0,
    },
  ],
]
`;
